addons/mass_mailing/tests/test_mailing_list.py

```diff
@@ -99,6 +99,18 @@ def test_mailing_contact_create(self):
             new = new.with_context(default_list_ids=[list_id])
             self.assertFalse(any(contact.opt_out for contact in new))
 
+        contact_form = Form(self.env['mailing.contact'])
+        contact_form.name = 'Contact_test'
+        with contact_form.subscription_list_ids.new() as subscription:
+            subscription.list_id = self.mailing_list_1
+            subscription.opt_out = True
+        with contact_form.subscription_list_ids.new() as subscription:
+            subscription.list_id = self.mailing_list_2
+            subscription.opt_out = False
+        contact = contact_form.save()
+        self.assertTrue(contact.subscription_list_ids[0].unsubscription_date)
```

Comment:
You could use `@freeze_time` and check that the date is set to whatever you freeze it with.

Comparing a date with True is a bit strange.